kafka-handler/src/java/org/apache/hadoop/hive/kafka/KafkaSerDe.java

```diff
@@ -369,6 +402,26 @@ private SubStructObjectInspector(StructObjectInspector baseOI, int toIndex) {
     }
   }
 
+    /**
+     * The converter reads bytes from kafka message and skip first @skipBytes from beginning.
+     *
+     * For example:
+     *       The Confluent Avro serializer adds 5 magic bytes that represents Schema ID as Integer to the message.
+     */
+  static class AvroSkipBytesConverter extends AvroBytesConverter {
+    private final int skipBytes;
+
+    AvroSkipBytesConverter(Schema schema, int skipBytes) {
+      super(schema);
+      this.skipBytes = skipBytes;
+    }
+
+    @Override
+    Decoder getDecoder(byte[] value) {
+      return DecoderFactory.get().binaryDecoder(value, this.skipBytes, value.length - this.skipBytes, null);
```

Comment:
Should there be validation that `skipBytes.length < value.length`?

Reply:
BinaryDecoder already throws a nice ArrayIndexOutOfBoundsException in this case, so I'm going to update to catch that, wrap in a SerDe exception, and keep going.